tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7843,6 +7841,8 @@ def err_hlsl_load_from_mesh_out_arrays: Error<
    "output arrays of a mesh shader can not be read from">;
 def err_hlsl_out_indices_array_incorrect_access: Error<
    "a vector in out indices array must be accessed as a whole">;
+def err_hlsl_unsupported_long_vector: Error<
+   "Vectors of over 4 elements in %0 are not supported">;
```

Comment:
Seems like the element count could be a format argument too so that we can error on vectors too large for the large-vectors proposal too?